src/converters/cover.ts

```diff
@@ -27,12 +27,28 @@ export class CoverCreator implements ServiceCreator {
           exposesHasFeatures(e) &&
           !accessory.isServiceHandlerIdKnown(CoverHandler.generateIdentifier(e.endpoint))
       )
-      .forEach((e) => this.createService(e as ExposesEntryWithFeatures, accessory));
+      .forEach((e) => {
+        let exposesMotorState: boolean = false;
+        const motorStateExpose = exposes.find(
+          (m) =>
+            CoverHandler.generateIdentifier(m.endpoint) === CoverHandler.generateIdentifier(e.endpoint) &&
+            m.type === ExposesKnownTypes.ENUM &&
+            m.name === 'motor_state' &&
+            exposesHasEnumProperty(m) &&
+            m.values.includes(CoverHandler.MOTOR_STATE_OPENING) &&
+            m.values.includes(CoverHandler.MOTOR_STATE_CLOSING) &&
+            m.values.includes(CoverHandler.MOTOR_STATE_STOPPED)
+        ) as ExposesEntry[] | undefined;
+        if (motorStateExpose !== undefined) {
+          exposesMotorState = true;
+        }
+        this.createService(e as ExposesEntryWithFeatures, accessory, exposesMotorState);
```

Comment:
You should pass the `motorStateExpose` (as `ExposesEntryWithEnumProperty`) instead of just a boolean.
You need that object to get the key (`motorStateExpose.property`) you need to check for in the `state` you receive.

Instead of the boolean property `modern` in the `CoverHandler`, you'd have an `private readonly motorStateExpose: ExposesEntryWithEnumProperty | undefined`.

Don't forget to also add this new property to the logic of `getableKeys()`.